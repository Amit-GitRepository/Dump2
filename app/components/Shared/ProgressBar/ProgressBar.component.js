import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './ProgressBar.styles';
import {colors} from '../../../themes/constants.styles';
import {View} from 'react-native';

class ProgressBar extends Component {
  render () {
    const {percentage, fillColor, backgroundColor} = this.props;
    const fillContainer = {
      backgroundColor: backgroundColor
    };
    const fillBar = {
      flex: percentage,
      backgroundColor: fillColor
    };
    const percentageValue = (percentage > 100) ? 100 : percentage;
    const emptyFlex = {
      flex: 100 - percentageValue
    };
    return (
      <View style={[fillContainer, styles.container]}>
        {percentageValue ? <View style={[fillBar, styles.progressFill]}/> : null}
        <View style={[emptyFlex, styles.progressEmpty]}/>
      </View>
    );
  }
}

ProgressBar.defaultProps = {
  percentage: 100,
  fillColor: colors.PRIMARY_ACTIONABLE,
  backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST
};

ProgressBar.propTypes = {
  percentage: PropTypes.number.isRequired,
  fillColor: PropTypes.string,
  backgroundColor: PropTypes.string
};

export default ProgressBar;
