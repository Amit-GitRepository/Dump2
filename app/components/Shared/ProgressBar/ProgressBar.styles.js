import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    borderWidth: 0,
    flexDirection: 'row',
    borderRadius: 4
  },
  progressFill: {
    paddingVertical: 5,
    borderRadius: 4
  },
  progressEmpty: {
    paddingVertical: 5,
    borderTopRightRadius: 4,
    borderBottomRightRadius: 4
  }
});
