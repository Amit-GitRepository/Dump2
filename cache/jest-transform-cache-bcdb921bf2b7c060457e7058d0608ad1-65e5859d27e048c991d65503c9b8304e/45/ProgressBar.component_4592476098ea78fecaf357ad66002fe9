Object.defineProperty(exports,"__esModule",{value:true});var cov_bpj48crfr=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/ProgressBar/ProgressBar.component.js',hash='a46aa3c476c06b330022b6cb16f9ea39cfaaabdc',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/ProgressBar/ProgressBar.component.js',statementMap:{'0':{start:{line:9,column:53},end:{line:9,column:63}},'1':{start:{line:10,column:26},end:{line:12,column:5}},'2':{start:{line:13,column:20},end:{line:16,column:5}},'3':{start:{line:17,column:28},end:{line:17,column:65}},'4':{start:{line:18,column:22},end:{line:20,column:5}},'5':{start:{line:21,column:4},end:{line:26,column:6}},'6':{start:{line:30,column:0},end:{line:34,column:2}},'7':{start:{line:36,column:0},end:{line:40,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:8,column:2},end:{line:8,column:3}},loc:{start:{line:8,column:12},end:{line:27,column:3}},line:8}},branchMap:{'0':{loc:{start:{line:17,column:28},end:{line:17,column:65}},type:'cond-expr',locations:[{start:{line:17,column:49},end:{line:17,column:52}},{start:{line:17,column:55},end:{line:17,column:65}}],line:17},'1':{loc:{start:{line:23,column:9},end:{line:23,column:80}},type:'cond-expr',locations:[{start:{line:23,column:27},end:{line:23,column:73}},{start:{line:23,column:76},end:{line:23,column:80}}],line:23}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},f:{'0':0},b:{'0':[0,0],'1':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _ProgressBar=require('./ProgressBar.styles');var _ProgressBar2=_interopRequireDefault(_ProgressBar);
var _constants=require('../../../themes/constants.styles');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

ProgressBar=function(_Component){_inherits(ProgressBar,_Component);function ProgressBar(){_classCallCheck(this,ProgressBar);return _possibleConstructorReturn(this,(ProgressBar.__proto__||Object.getPrototypeOf(ProgressBar)).apply(this,arguments));}_createClass(ProgressBar,[{key:'render',value:function render()
{cov_bpj48crfr.f[0]++;var _ref=(cov_bpj48crfr.s[0]++,
this.props),percentage=_ref.percentage,fillColor=_ref.fillColor,backgroundColor=_ref.backgroundColor;
var fillContainer=(cov_bpj48crfr.s[1]++,{
backgroundColor:backgroundColor});

var fillBar=(cov_bpj48crfr.s[2]++,{
flex:percentage,
backgroundColor:fillColor});

var percentageValue=(cov_bpj48crfr.s[3]++,percentage>100?(cov_bpj48crfr.b[0][0]++,100):(cov_bpj48crfr.b[0][1]++,percentage));
var emptyFlex=(cov_bpj48crfr.s[4]++,{
flex:100-percentageValue});cov_bpj48crfr.s[5]++;

return(
_react2.default.createElement(_reactNative.View,{style:[fillContainer,_ProgressBar2.default.container]},
percentageValue?(cov_bpj48crfr.b[1][0]++,_react2.default.createElement(_reactNative.View,{style:[fillBar,_ProgressBar2.default.progressFill]})):(cov_bpj48crfr.b[1][1]++,null),
_react2.default.createElement(_reactNative.View,{style:[emptyFlex,_ProgressBar2.default.progressEmpty]})));


}}]);return ProgressBar;}(_react.Component);cov_bpj48crfr.s[6]++;


ProgressBar.defaultProps={
percentage:100,
fillColor:_constants.colors.PRIMARY_ACTIONABLE,
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST};cov_bpj48crfr.s[7]++;


ProgressBar.propTypes={
percentage:_propTypes2.default.number.isRequired,
fillColor:_propTypes2.default.string,
backgroundColor:_propTypes2.default.string};exports.default=


ProgressBar;