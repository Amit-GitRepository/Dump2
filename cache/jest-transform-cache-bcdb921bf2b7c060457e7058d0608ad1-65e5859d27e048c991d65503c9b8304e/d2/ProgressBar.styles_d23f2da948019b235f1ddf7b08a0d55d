Object.defineProperty(exports,"__esModule",{value:true});var cov_1tehzidb4t=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/ProgressBar/ProgressBar.styles.js',hash='f599fcce2d0664c8bf89894a6b40a757f8765fb2',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/ProgressBar/ProgressBar.styles.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _reactNative=require('react-native');exports.default=

_reactNative.StyleSheet.create({
container:{
borderWidth:0,
flexDirection:'row',
borderRadius:4},

progressFill:{
paddingVertical:5,
borderRadius:4},

progressEmpty:{
paddingVertical:5,
borderTopRightRadius:4,
borderBottomRightRadius:4}});