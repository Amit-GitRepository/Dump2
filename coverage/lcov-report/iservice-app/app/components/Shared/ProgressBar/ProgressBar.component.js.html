<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Shared/ProgressBar/ProgressBar.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Shared/ProgressBar</a> ProgressBar.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './ProgressBar.styles';
import {colors} from '../../../themes/constants.styles';
import {View} from 'react-native';
&nbsp;
class ProgressBar extends Component {
  render () {
    const {percentage, fillColor, backgroundColor} = this.props;
    const fillContainer = {
      backgroundColor: backgroundColor
    };
    const fillBar = {
      flex: percentage,
      backgroundColor: fillColor
    };
    const percentageValue = (percentage &gt; 100) ? 100 : percentage;
    const emptyFlex = {
      flex: 100 - percentageValue
    };
    return (
      &lt;View style={[fillContainer, styles.container]}&gt;
        {percentageValue ? &lt;View style={[fillBar, styles.progressFill]}/&gt; : <span class="branch-1 cbranch-no" title="branch not covered" >null}</span>
        &lt;View style={[emptyFlex, styles.progressEmpty]}/&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
ProgressBar.defaultProps = {
  percentage: 100,
  fillColor: colors.PRIMARY_ACTIONABLE,
  backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST
};
&nbsp;
ProgressBar.propTypes = {
  percentage: PropTypes.number.isRequired,
  fillColor: PropTypes.string,
  backgroundColor: PropTypes.string
};
&nbsp;
export default ProgressBar;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
