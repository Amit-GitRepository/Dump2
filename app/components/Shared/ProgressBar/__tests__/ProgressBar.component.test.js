import ProgressBar from '../ProgressBar.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('ProgressBar component', () => {
  it('renders correctly', () => {
    const iconTree = renderer.create(<ProgressBar />).toJSON();
    expect(iconTree).toMatchSnapshot();
  });
  it('renders correctly with greater than 100 value also', () => {
    const wrapper = shallow(<ProgressBar percentage={120}/>);
    expect(wrapper).toBeDefined();
  });
});
