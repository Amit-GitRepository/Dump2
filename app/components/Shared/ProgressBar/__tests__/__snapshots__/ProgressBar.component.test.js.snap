// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Object {
        "borderRadius": 4,
        "borderWidth": 0,
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F93E60",
          "flex": 100,
        },
        Object {
          "borderRadius": 4,
          "paddingVertical": 5,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "borderBottomRightRadius": 4,
          "borderTopRightRadius": 4,
          "paddingVertical": 5,
        },
      ]
    }
  />
</View>
`;
